src/components/ErrorBoundary.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import { Typography, Button, Collapse } from '@mui/material';
+import emoji from '../assets/errorboundary-emoji.png';
+
+type ErrorBoundaryProps = {
+  children: React.ReactNode;
+};
+
+type ErrorBoundaryState = {
+  hasError: boolean;
+  error?: Error | null;
+  errorInfo?: React.ErrorInfo | null;
+  showDetails: boolean;
+};
+
+class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
+  constructor(props: ErrorBoundaryProps) {
+    super(props);
+    this.state = { hasError: false, error: null, errorInfo: null, showDetails: false };
+  }
```

Comment:
Why does this have to be a class component and not a function component like everything else? That seems interesting and unexpected. So assuming that there is a good reason for it, please explain it above in a brief comment so we don't have to wonder every time we come back and read this in the future

Reply:
Ah this was done a while ago in query tool, let me go do some digging.